app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -183,6 +184,12 @@ public void addSupport(
         @NotNull ModifiableModelsProvider modifiableModelsProvider) {
       FacetType<AppEngineFlexibleFacet, AppEngineFlexibleFacetConfiguration> facetType =
           AppEngineFlexibleFacet.getFacetType();
+      // pre-condition - multiple facets of the same type are not allowed and result in a critical
+      // error, check for existing facet explicitly and do nothing if it was already added.
+      if (AppEngineUtil.isAnyAppEngineFacetAlreadyAdded(module)) {
```

Comment:
this checks for multiple facets of different types right? This contradicts the comment I think.

Reply:
Right, thanks, I can fix the comment. For now, the behavior is we don't allow more than one facet, and we cannot combine standard and flex, is this correct from App Engine point of view?

Reply:
From our plugin's point of view we only want to allow a single facet (despite it being possible to have an app configured for both flex and standard).

Reply:
Awesome, thanks.